optuna/_convert_positional_args.py

```diff
@@ -37,9 +53,13 @@ def converter_decorator(func: "Callable[_P, _T]") -> "Callable[_P, _T]":
 
         @wraps(func)
         def converter_wrapper(*args: Any, **kwargs: Any) -> "_T":
-            if len(args) >= 1:
+            positional_arg_names = _get_positional_arg_names(func)
+            inferred_args = _infer_given_args(previous_positional_arg_names, *args)
+            if len(inferred_args) > len(positional_arg_names):
+                kwargs_expected = set(inferred_args) - set(positional_arg_names)
                 warnings.warn(
-                    f"{func.__name__}(): Please give all values as keyword arguments."
+                    f"{func.__name__}() got {kwargs_expected} as positional arguments "
+                    "but they were expected to be given as keyword arguments."
                     " See https://github.com/optuna/optuna/issues/3324 for details.",
```

Comment:
It's almost LGTM. How about removing this line because this error is not only related to #3324?

Reply:
Right, because of the last change, #3324 does not relate to this line anymore. (while it still makes sense to use kwargs only, the future warning for it is not the default Python design, so I removed it and stick to your suggestion.)